src/npcmove.cpp

```diff
@@ -3711,14 +3723,18 @@ void npc::use_painkiller()
 static float rate_food( const item &it, int want_nutr, int want_quench )
 {
     const auto &food = it.get_comestible();
+
+    // Don't eat it if it's not food.
     if( !food ) {
         return 0.0f;
     }
 
+    // Don't eat it if it's filled with parasites
     if( food->parasites && !it.has_flag( flag_NO_PARASITES ) ) {
         return 0.0;
     }
 
+    // QUESTION: Will this return the *actual* (not default) nutrients for player-cooked food?
```

Comment:
This returns nutrition based on `itype`, ignoring ingredients. I think that should be fixed or at least commented with TODO
Afaik, there is Character::get_kalorie_value or something close that recursively calculates base don ingredients.

Reply:
Oh, thank you! `Character::nutrition_for()` and `Character::compute_effective_nutrients()` seem to do what we want.

There's also `Character::fun_for()` that's similar for fun.